code/backend/batch/utilities/search/Search.py

```diff
@@ -0,0 +1,13 @@
+from ..search.AzureSearchHandler import AzureSearchHandler
+from ..search.IntegratedVectorizationSearchHandler import (
+    IntegratedVectorizationSearchHandler,
+)
+
+
+class Search:
```

Comment:
I really like this. I think in general we should hide all the IV/non-IV if-else logic behind factory classes and handlers like you did here.